src/thumbnailer_slope_optim.cc

```diff
@@ -30,23 +30,21 @@ Thumbnailer::Status Thumbnailer::GenerateAnimationSlopeOptim(
   // If there is no frame encoded in near-lossless, use
   // GenerateAnimationEqualQuality() to generate animation.
   int num_near_ll_frames = 0;
-  for (auto& frame : frames_)
-    if (frame.near_lossless) {
-      ++num_near_ll_frames;
-    }
-  if (!num_near_ll_frames) {
-    CHECK_THUMBNAILER_STATUS(GenerateAnimationEqualQuality(webp_data));
-    return kOk;
+  for (const auto& frame : frames_) {
+    if (frame.near_lossless) ++num_near_ll_frames;
   }
-
-  int curr_anim_size = webp_data->size;
-  const int KMaxIter = 5;
-  for (int i = 0; i < KMaxIter; ++i) {
-    CHECK_THUMBNAILER_STATUS(LossyEncodeNoSlopeOptim(webp_data));
-    if (curr_anim_size == webp_data->size) break;
-    curr_anim_size = webp_data->size;
+  if (num_near_ll_frames == 0) {
```

Comment:
If you keep the code like this, please add a comment here to clearly explain this choice. Apart from that the PR looks good to me.